### ISSUE TITLE
Qwen2 VL (pretrain_qwen.py) pad logic error

### ISSUE THREAD

**Issue by jason9693** [alibaba/Pai-Megatron-Patch#448]
When try to make [ltor_mask](https://github.com/alibaba/Pai-Megatron-Patch/blob/3b90f6832a6645d47aacc3583317b7dd64c1ce67/examples/qwen2_vl/pretrain_qwen.py#L284), 

the `attention mask` is generated as follows:
```python
get_rope_index(
        input_ids=input_ids,
        image_grid_thw=image_thw_grids,
        video_grid_thw=video_thw_grids,
        attention_mask=target == pad_token
    )

```

However, according to the documentation for the get_rope_index function, the attention_mask parameter (a torch.Tensor of shape (batch_size, sequence_length), optional) is described as follows:

```
attention_mask (`torch.Tensor` of shape `(batch_size, sequence_length)`, *optional*):
            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:

            - 1 for tokens that are **not masked**,
            - 0 for tokens that are **masked**.
```

Based on this logic, it seems that only the pad_token is marked as unmasked. Could you please confirm if this logic is correct?

**Comment by lostkevin** [comment]
Thanks for pointing out the issue. I've create #449 to fix this problem. Now the code will label all non-pad token (in Qwen2-VL, <end_of_text>, which should not appear in the chat template) as not masked.

```
get_rope_index(
        input_ids=input_ids,
        image_grid_thw=image_thw_grids,
        video_grid_thw=video_thw_grids,
        attention_mask=input_ids != pad_token
    )
```
